### DISCUSSION TITLE
export APM traces from Weblogic Server/OSB12.2.4.0

### DISCUSSION THREAD

**Discussion by krishna-yalla** [open-telemetry/opentelemetry-java-instrumentation#6226] [Q&A]
Does anyone have any idea how to export APM-Application Performance Management traces from Weblogic Server using opentelemetry-java-instrumentation?

We are doing a POC to collect APM traces into our opensource backend ( (Prometheus, Loki, and Tempo)opentelemetry-java agent which is enabling the APM traceID as part of the auto instrumentation.

However traceID is not getting written in to any weblogic logger file( .adm,.out,etc).we are looking for some help or any direction how the opentelemetry-java-instrumentation generates APM TraceID and writes into Logger file.

Do we need to configure the MDC logger or any other configuration to achieve this?

Product Details :
Weblogic 12.2.1.4.0
OSB 12.2.1.4.0
Java 8
Linux version: RHEL 7.9

**Comment by mateuszrzeszutek** [comment]
Hey @krishna-yalla ,
The javaagent will inject all the data documented in [this doc](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/logger-mdc-instrumentation.md#logger-mdc-auto-instrumentation) to the logger MDC, but it won't change the logging output format. You have to change the configuration of your logger and make sure that it prints out the MDC values (like in the spring boot example on the page I linked).
